sudoku_solver/sudoku_solver.cpp

```diff
@@ -0,0 +1,101 @@
+#include <bits/stdc++.h>
+using namespace std;
+
+bool isPossible(int arr[][9],int r,int c,int n)
+{
+  for(int i=0;i<9;i++)
+  {
+    if(arr[r][i]==n)
+      return false;
+    if(arr[i][c]==n)
+      return false;
+  }
+  int sr=(r/3)*3;
+  int sc=(c/3)*3;
+  for(int i=sr;i<sr+3;i++)
+  {
+    for(int j=sc;j<sc+3;j++)
+    {
+      if(arr[i][j]==n)
+        return false;
+    }
+  }
+  return true;
+}
+
+bool solve(int arr[][9],int r,int c)
+{
+  if(r==9)
+    return true;
+  if(c==9)
+    return solve(arr,r+1,0);
+  if(arr[r][c]==0)
+  {
+    for(int i=1;i<=9;i++)
+    {
+      if(isPossible(arr,r,c,i))
+      {
+        arr[r][c]=i;
+        bool ans=solve(arr,r,c+1);
+        if(ans)
+          return ans;
+        arr[r][c]=0;
+      }
+    }
+    return false;
+  }
+  else
+    return solve(arr,r,c+1);  
+}
+
+bool sudokuSolver(int board[][9])
+{
+ return solve(board,0,0); 
+}
+
+void printSudoku(int sudoku[][9])
+{
+	cout<<"solved sudoku is : "<<endl;
+	for(int i=0;i<9;i++)
+	{
+		for(int j=0;j<9;j++)
+			cout<<sudoku[i][j]<<" ";
+		cout<<endl;
+	}
+}
+
+int main()
+{
+	int sudoku[9][9];
```

Comment:
better to initialse 2d array with 0

Reply:
Rest LGTM 👍

Reply:
can you please tell why it would be better to initialize the array with zero?

Reply:
It is a better practice to avoid garbage values in your code as much as you can,